app/src/main/java/org/schabi/newpipe/streams/io/StoredDirectoryHelper.java

```diff
@@ -52,47 +52,45 @@ public StoredDirectoryHelper(@NonNull Context context, @NonNull Uri path, String
             throw new IOException(e);
         }
 
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP)
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
             throw new IOException("Storage Access Framework with Directory API is not available");
+        }
 
         this.docTree = DocumentFile.fromTreeUri(context, path);
 
-        if (this.docTree == null)
+        if (this.docTree == null) {
             throw new IOException("Failed to create the tree from Uri");
+        }
     }
 
-    @TargetApi(Build.VERSION_CODES.KITKAT)
-    public StoredDirectoryHelper(@NonNull URI location, String tag) {
-        ioTree = new File(location);
-        this.tag = tag;
-    }
-
-    public StoredFileHelper createFile(String filename, String mime) {
+    public StoredFileHelper createFile(final String filename, final String mime) {
         return createFile(filename, mime, false);
     }
 
-    public StoredFileHelper createUniqueFile(String name, String mime) {
-        ArrayList<String> matches = new ArrayList<>();
-        String[] filename = splitFilename(name);
-        String lcFilename = filename[0].toLowerCase();
+    public StoredFileHelper createUniqueFile(final String name, final String mime) {
+        final ArrayList<String> matches = new ArrayList<>();
+        final String[] filename = splitFilename(name);
+        final String lcFilename = filename[0].toLowerCase();
 
         if (docTree == null || Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
-            for (File file : ioTree.listFiles())
+            for (final File file : ioTree.listFiles()) {
                 addIfStartWith(matches, lcFilename, file.getName());
+            }
         } else {
             // warning: SAF file listing is very slow
-            Uri docTreeChildren = DocumentsContract.buildChildDocumentsUriUsingTree(
-                    docTree.getUri(), DocumentsContract.getDocumentId(docTree.getUri())
-            );
+            final Uri docTreeChildren = DocumentsContract.buildChildDocumentsUriUsingTree(
+                    docTree.getUri(), DocumentsContract.getDocumentId(docTree.getUri()));
 
-            String[] projection = new String[]{COLUMN_DISPLAY_NAME};
-            String selection = "(LOWER(" + COLUMN_DISPLAY_NAME + ") LIKE ?%";
-            ContentResolver cr = context.getContentResolver();
+            final String[] projection = new String[]{COLUMN_DISPLAY_NAME};
+            final String selection = "(LOWER(" + COLUMN_DISPLAY_NAME + ") LIKE ?%";
+            final ContentResolver cr = context.getContentResolver();
 
-            try (Cursor cursor = cr.query(docTreeChildren, projection, selection, new String[]{lcFilename}, null)) {
+            try (Cursor cursor = cr.query(docTreeChildren, projection, selection,
```

Comment:
Can `final` be put here?
```suggestion
            try (final Cursor cursor = cr.query(docTreeChildren, projection, selection,
```

Reply:
No, it's `final` by default, and thus Checkstyle would complain about it being redundant